lstchain/scripts/lstchain_add_source_dependent_parameters.py

```diff
@@ -61,11 +62,17 @@ def main():
             pass
 
     dl1_params = pd.read_hdf(dl1_filename, key=dl1_params_lstcam_key)
-    subarray_info = SubarrayDescription.from_hdf(dl1_filename)
-    tel_id = config["allowed_tels"][0] if "allowed_tels" in config else 1
-    focal_length = subarray_info.tel[tel_id].optics.equivalent_focal_length
-
-    src_dep_df = pd.concat(get_source_dependent_parameters(dl1_params, config, focal_length=focal_length), axis=1)
+    try:
+        subarray_info = SubarrayDescription.from_hdf(dl1_filename)
+        tel_id = config["allowed_tels"][0] if "allowed_tels" in config else 1
+        effective_focal_length = subarray_info.tel[tel_id].optics.effective_focal_length
+    except OSError:
+        print("subarray table is not readable because of the version inompatibility.")
+        print("Use the effective focal lentgh for the standard LST optics")
+        effective_focal_length = OPTICS.effective_focal_length
+        
+    src_dep_df = pd.concat(get_source_dependent_parameters(
+        dl1_params, config, effective_focal_length=effective_focal_length), axis=1)
```

Comment:
`src_x` and `src_y` is updated using effective focal length here (I thought this was done before `get_expected_source_pos` function, but it is not the case... I will update the code) 
https://github.com/cta-observatory/cta-lstchain/blob/500d7b2e15596c532a67535c369e10d379d0e097/lstchain/reco/dl1_to_dl2.py#L629

Reply:
As I said, I think you have to update get_expected_source_pos so that it recalculates the position, not just propagates...

Reply:
OK, it is true. Then I will modify codes to always recalculate the source position inside this function.
This function is only used for source-dep analysis. But, it would be better to recalculate the source position somewhere else also for the source-indep analysis addressing this comment, correct?
https://github.com/cta-observatory/cta-lstchain/pull/1131#issuecomment-1633677178

Reply:
Ok, I see.

For `lstchain_add_source_dependent_parameters.py`, I'd even remove this script since source-dep parameters are also computed during dl1_to_dl2 step if src-dep parameters don't exist.
This script overwrites the existing DL1 data to add source-dependent data frame, so it doesn't make sense for the current data analysis scheme since we analyze the common DL1 data produced by OSA.